OMCompiler/Compiler/NFFrontEnd/NFTyping.mo

```diff
@@ -549,7 +549,7 @@ algorithm
         (exp, ty, var, purity) := typeExp(range, InstContext.set(context, NFInstContext.ITERATION_RANGE), info);
 
         // If the iteration range is structural, it must be a parameter expression.
-        if structural and var > Variability.PARAMETER then
+        if structural and var > Variability.PARAMETER and Flags.isSet(Flags.NF_SCALARIZE) then
```

Comment:
This seems too permissive, it should probably be at least `(var == Variability.NON_STRUCTURAL_PARAMETER and Flags.isSet(Flags.NF_SCALARIZE))`.

Reply:
i changed it to `(Flags.isSet(Flags.NF_SCALARIZE) or not var == Variability.NON_STRUCTURAL_PARAMETER)`. The error only gets surpressed if we are not scalarizing and it's a non structural parameter.

Maybe there is a prettier way to write this?

Reply:
Yeah, sorry, I meant `var > Variability.NON_STRUCTURAL_PARAMETER and Flags.isSet(Flags.NF_SCALARIZE)`, but your way works too of course. I'd check the variability first since checking the flag is much more expensive.